Source/Core/DolphinQt2/Resources.cpp

```diff
@@ -2,46 +2,69 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <algorithm>
+
+#include <QGuiApplication>
+#include <QScreen>
 #include <QStringList>
 
 #include "Common/CommonPaths.h"
 #include "Common/FileUtil.h"
 #include "DolphinQt2/Resources.h"
+#include "DolphinQt2/Settings.h"
 
 QList<QPixmap> Resources::m_platforms;
 QList<QPixmap> Resources::m_countries;
 QList<QPixmap> Resources::m_ratings;
 QList<QPixmap> Resources::m_misc;
 
+QString Resources::GetPreferredIconScale(int max)
+{
+  const auto dpr = QGuiApplication::primaryScreen()->devicePixelRatio();
+  const auto scale = std::min(dpr <= 1 ? 1 : (dpr <= 2) ? 2 : 4, max);
+
+  if (scale > 1)
+    return QStringLiteral("@%1x").arg(scale);
+  else
+    return QStringLiteral("");
+}
+
+QPixmap Resources::GetScaledIcon(const QString& name, int max, const QString& dir)
```

Comment:
This returns a pixmap, its name should reflect that rather than sounding like it returns an icon.